gpii/node_modules/flowManager/src/FlowManagerUtilities.js

```diff
@@ -62,20 +58,39 @@
      * @return (undefined) - function is asynchronous and doesn't return anything. Instead the event
      *      is fired with the modified mmpayload.
      */
-    gpii.request.flowManager.getSolutions = function (solutionsRegistryDataSource, deviceContext, event) {
+    gpii.flowManager.getSolutions = function (solutionsRegistryDataSource, deviceContext, event, onError) {
         var os = fluid.get(deviceContext, "OS.id"),
             version = fluid.get(deviceContext, "OS.version");
-        solutionsRegistryDataSource.get({
+        var promise = solutionsRegistryDataSource.get({
             os: os,
             version: version
-        }, function (solutions) {
+        });
+        promise.then(function (solutions) {
             var solutionsRegistryEntries = gpii.matchMakerFramework.filterSolutions(solutions, deviceContext);
             fluid.log("Fetched filtered solutions registry entries: " + JSON.stringify(solutionsRegistryEntries, null, 4));
             event.fire(solutionsRegistryEntries);
+        }, onError.fire);
+    };
+    
+    
+    gpii.flowManager.getDeviceContext = function (deviceReporterDataSource, event, errorEvent) {
+        console.log("getDeviceContext for dataSource with options ", deviceReporterDataSource.options);
+        var promise = deviceReporterDataSource.get();
+        promise.then(function (deviceData) {
+            console.log("Got deviceData ", deviceData);
+            if (deviceData.isError) {
+                errorEvent.fire({message: "Error in device reporter data: " + deviceData.message});
```

Comment:
Shall the function ```return``` after the ```errorEvent``` is fired to prevent it goes further to report an incorrect ```deviceData```?